src/python/WMCore/ReqMgr/CherryPyThreads/StatusChangeTasks.py

```diff
@@ -6,9 +6,76 @@
 
 from Utils.CherryPyPeriodicTask import CherryPyPeriodicTask
 from WMCore.Services.WMStats.WMStatsReader import WMStatsReader
+from WMCore.Services.WorkQueue.WorkQueue import WorkQueue
 from WMCore.Services.RequestDB.RequestDBWriter import RequestDBWriter
 from WMCore.Services.WMStats.DataStruct.RequestInfoCollection import RequestInfoCollection
 
+
+def moveForwardStatus(reqDBWriter, requestStatus):
+    
+    statusTransition = {"assigned": ["acquired", "running-open", "running-closed", "completed"], 
+                        "acquired": ["running-open", "running-closed", "completed"],
+                        "running-open": ["running-closed", "completed"],
+                        "aborted": ["completed"],
+                        "running-closed": ["completed"],
+                        "force-complete": ["completed"]}
+    
+    for status, nextStatus in statusTransition.items():
+        requests = reqDBWriter.getRequestByStatus([status])
+        for wf in requests:
+            stateFromGQ = requestStatus.get(wf, None)
+            # special case for failed workflow
+            if stateFromGQ is None:
+                continue
+            elif stateFromGQ == "failed" and status == "assigned":
+                reqDBWriter.updateRequestStatus(wf, stateFromGQ)
+                print("%s in %s moved to %s" % (wf, status, stateFromGQ))
+                continue
+            try:
+                i = nextStatus.index(stateFromGQ)
+            except ValueError:
+                # No state change needed
+                continue
+            # special case for aborted workflow - aborted-completed instead of completed
+            if status == "aborted" and i == 0:
+                reqDBWriter.updateRequestStatus(wf, "aborted-completed")
+                print("%s in %s moved to %s" % (wf, status, "aborted-completed"))
+            else:
+                for j in range(i+1):
+                    reqDBWriter.updateRequestStatus(wf, nextStatus[j])
+                    print("%s in %s moved to %s" % (wf, status, nextStatus[j]))
+    return
+
+def moveToArchivedForNoJobs(wmstatsService, reqDBWriter):
+    
+    statusTransition = {"aborted-completed": ["aborted-archived"],
+                            "rejected": ["rejected-archived"]}
+        
+    for status, nextStatusList in statusTransition.items():
+        
+        requests = wmstatsService.getRequestByStatus([status], 
+                                            jobInfoFlag = True, legacyFormat = True)
+        
+        print("checking %s workflows: %d" % (status, len(requests)))
+        
+        if len(requests) > 0:
+        
+            requestCollection = RequestInfoCollection(requests)
```

Comment:
BTW, archiving workflows seem to be much heavier than the other transitions since it has to get a bunch of information from wmstats. Maybe we should define a different polling cycle for archival (every hour or so).